src/PgNoticeMessagesPlugin.ts

```diff
@@ -0,0 +1,84 @@
+import { Plugin } from "graphile-build";
+import { OperationHookCallback, OperationHook } from "./OperationHooksPlugin";
+import { GraphQLResolveInfoWithMessages } from "./OperationMessagesPlugin";
+import * as assert from "assert";
+
+const PgNoticeMessagesPlugin: Plugin = function PgNoticeMessagesPlugin(
+  builder
+) {
+  builder.hook("init", (_, build) => {
+    const hookGenerator: OperationHookCallback = (fieldContext) => {
+      if (!fieldContext.scope.isRootMutation) {
+        // We absolutely cannot support isRootQuery because root query resolvers
+        // run in parallel, all using the same PG client, so we wouldn't know
+        // which field a notice applied to. We need the Postgres client to only
+        // do this one thing.
+        return null;
+      }
+
+      const makeProcessNotice = (
+        resolveInfo: GraphQLResolveInfoWithMessages
+      ) => (msg: any) => {
+        if (msg.code !== "OPMSG") {
+          // We only care about OPMSG NOTICEs
+          return;
+        }
+        // TODO
+        let json: any = null;
+        try {
+          json = JSON.parse(msg.detail);
+        } catch (e) {
+          console.dir(msg);
+          console.error("Failed to parse above OPMSG NOTICE from PostgreSQL");
+          console.error(e);
+        }
+        resolveInfo.graphileMeta.messages.push({
+          level: "info",
+          message: msg.message,
+          ...json,
+        });
+      };
+      let processNotice: ((msg: any) => void) | null = null;
+
+      const registerNotifyListener: OperationHookCallback = (
+        input,
+        _args,
+        context,
+        resolveInfo: GraphQLResolveInfoWithMessages
+      ) => {
+        assert(!processNotice, "processNotice should not be set yet!");
```

Comment:
This assertion passes on the first request, but fails on subsequent ones. Commenting it out does not seem to have any ill effects, and makes everything work as expected.

Reply:
Fixed :+1: